liquidctl/driver/gigabyte_rgb_fusion.py

```diff
@@ -0,0 +1,328 @@
+        'flash':                            (_FLASH_SPEEDS),
+        'double-flash':                     (_DOUBLE_FLASH_SPEEDS),
+        'color-cycle':                      (_COLOR_CYCLE_SPEEDS),    
+    }
+
+    def __init__(self, device, description, speed_channel_count, color_channel_count, **kwargs):
+        """Instantiate a driver with a device handle."""
+        speed_channels = {'fan{}'.format(i + 1): (1 << i)
+                          for i in range(speed_channel_count)}
+        color_channels = {
+            'IOLED':  (0x20, 0x01),
+            'LED1':   (0x21, 0x02),
+            'PCHLED': (0x22, 0x04),
+            'PCILED': (0x23, 0x08),
+            'LED2':   (0x24, 0x10),
+            'DLED1':  (0x25, 0x20),
+            'DLED2':  (0x26, 0x40),
+        }
+        super().__init__(device, description, speed_channels, color_channels, **kwargs)
+
+    def initialize(self, **kwargs):
+        """Initialize the device.
+
+        Detects and reports all connected fans and LED accessories, and allows
+        subsequent calls to get_status.
+
+        Returns a list of (key, value, unit) tuples.
+        """
+        self.device.clear_enqueued_reports()
+        status = []
+        # initialize
+        self._write_single(0x60) # 0x60 = Initialize code
+        data=self._get_feature_report(self._REPORT_ID)
+        if data[0]==0xcc and data[1]==0x01:
+            num_devices = data[3]
+            ver_major = data[4]
+            ver_minor = data[5]
+            ver_build = data[6]
+            ver_sub = data[7]
+            index = 12  # first letter of device name in 'data'
+            dev_name = ""
+            while data[index] != 0 and index < self._READ_LENGTH:
+                dev_name += chr(data[index])
+                index += 1
+
+            status.append(('Name', dev_name, ''))
+            status.append(('Version', '{}.'.format(ver_major)+'{}.'.format(ver_minor)+
+                    '{}.'.format(ver_build)+'{}'.format(ver_sub), ''))
+            status.append(('LED channels', num_devices, ''))
+        self.device.release()
+        return status
+
+    def _write_one_cycle(self):
+        self._write_series()
+        self._write_end_block()
+        
+    def _write_single(self, code):
+        self._send_feature_report([self._REPORT_ID, code])
+        
+    def _write_series(self):
+        """Send a series of initializer data packets"""
+        for x in range(0x20,0x28):
+            self._send_feature_report([self._REPORT_ID, x])
+            
+    def _write_end_block(self):
+        self._send_feature_report([self._REPORT_ID, 0x28, 0xff])
+
+    def get_status(self, **kwargs):
+        """Get a status report.
+
+        Returns a list of (key, value, unit) tuples.
+        """
+        if not self._speed_channels:
+            return []
+        status = []
+        status.append(('Lighting channels only. Nothing to report', '', ''))
+        return sorted(status)
```

Comment:
You can simply unconditionally return `[]` if there is never any data to report.

Reply:
To be honest, this driver was not ready for a Pull Request. The pull request was issued so that a Windows executable would be created.

Reply:
Yeah, I figured that.  It's just that it would have been reasonable for you to think that `speed_channel_count` was mandatory, based on the other driver you developed, so I though it could be useful for you to know that you can simply this.